OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/cg/AndroidICCAnalysis.scala

```diff
@@ -0,0 +1,1413 @@
+            this.pc = pc
+            calledMethods ++= callee.findMethod(onReceive)
+        } else if (serviceMethods.contains(nme)) {
+            this.pc = pc
+            nme match {
+                case `startService` ⇒ calledMethods ++= callee.findMethod(onCreate) ++ callee.findMethod(onStartCommand)
+                case `bindService` ⇒ calledMethods ++= callee.findMethod(onCreate) ++ callee.findMethod(onBind) ++
+                    callee.findMethod(onRebind)
+                case `stopService`   ⇒ calledMethods ++= callee.findMethod(stopSelf)
+                case `unbindService` ⇒ calledMethods ++= callee.findMethod(onUnbind)
+            }
+        }
+        calledMethods
+    }
+}
+
+class intentFilter() {
+    final val addDataAuthority = "addDataAuthority"
+    final val addDataPath = "addDataPath"
+    final val addDataScheme = "addDataScheme"
+    final val addDataSSP = "addDataSchemeSpecificPart"
+    final val addDataType = "addDataType"
+    final val paramRegex = ".*\\(\"|\"\\).*"
+    final val addAction = "addAction"
+    final val addCategory = "addCategory"
+    final val registerReceiver = "registerReceiver"
+    final val receiverRegex = ".*\\: |\\[.*"
+
+    final val filterDataMethods: List[String] = List(addDataAuthority, addDataPath,
+        addDataScheme, addDataSSP, addDataType)
+
+    var receiver: ClassFile = _
+    var componentType: String = _
+    var registeredReceivers: ListBuffer[ObjectType] = ListBuffer.empty[ObjectType]
+
+    var actions: ListBuffer[String] = ListBuffer.empty[String]
+    var categories: ListBuffer[String] = ListBuffer.empty[String]
+
+    var dataTypes: ListBuffer[String] = ListBuffer.empty[String]
+    var dataAuthorities: ListBuffer[String] = ListBuffer.empty[String]
+    var dataPaths: ListBuffer[String] = ListBuffer.empty[String]
+    var dataSchemes: ListBuffer[String] = ListBuffer.empty[String]
+    var dataSSPs: ListBuffer[String] = ListBuffer.empty[String]
+
+    def addAction(action: String): Unit = {
+        actions += action
+
+    }
+    def addCategory(category: String): Unit = {
+        categories += category
+    }
+    def addDataType(dataType: String): Unit = {
+        dataTypes += dataType
+    }
+
+    def setReceiver(rec: ClassFile): Unit = {
+        receiver = rec
+    }
+
+    def cloneFilter(): intentFilter = {
+        val clone = new intentFilter()
+        clone.actions = actions
+        clone.categories = categories
+        clone.dataAuthorities = dataAuthorities
+        clone.dataTypes = dataTypes
+        clone.dataPaths = dataPaths
+        clone.dataSchemes = dataSchemes
+        clone.componentType = componentType
+        clone.receiver = receiver
+        clone
+    }
+
+    def evaluateUseSites(
+                          tacCode: AITACode[TACMethodParameter, ValueInformation],
+                          useSites: List[Int]
+    ): Unit = {
+        val statements = tacCode.stmts
+        useSites.foreach { ln ⇒
```

Comment:
What is "ln" supposed to mean? As you're iterating over useSites, I'd usually expect the variable here to be called "use", though somethig more specific may also be used to make things clearer.